ecommerce-ui/src/BnbListings/BnbListings.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Link
+} from "react-router-dom";
+import './BnbListings.css';
+import SuperHostIcon from '../HostIcon/SuperHostIcon';
+import HostIcon from '../HostIcon/HostIcon';
+
+class BnbListings extends Component {
+
+  getHostIcon() {
+    return (
+      <div className="host-icon">
+        <HostIcon />
+      </div>
+    );
+  };
+
+  getSuperHostIcon() {
+    return (
+      <div className="host-icon">
+        <SuperHostIcon />
+      </div>
+    );
+  };
+
+  getPaymentDescription(descript) {
+    return (
+      <span>
+        · {descript}
+      </span>
+    );
+  };
+
+  render() {
+    const isSuperHost = this.props.bnb.host.isSuperhost;
+    const image = this.props.bnb.image;
+    const backgroundMediaStyles = {
+        backgroundImage: 'url(' + image + ')',
+    };
+    const starNumber = this.props.bnb.rating.stars;
+    const paymentDescription = this.props.bnb.payment.description;
+    return (
+      <section className="bnb-listing">
+        <Link to={`/id/${this.props.bnb.id}`} style={{ textDecoration: 'none' }}>
```

Comment:
The test for this component fails because it says that "You should not use `Link` outside a `Router`." If I add a `BrowserRouter` to this component the test passes but my App breaks. There are already `BrowserRouter` tags surrounding the parent component. 

Additionally I don't know for certain what the test is referring to with this message "Warning: Failed context type: The context `router` is marked as required in `Link`, but its value is `undefined`." It tells me the line number as being where `Link` is used (the line above). But there is no component that I import called `router`, only `BrowserRouter`. I've tried to search dependancies but I'm not seeing anything.

Reply:
I have updated BnbListings.test.js to add MemoryRouter as we did in class and that has fixed this error.